lib/gitsh/tab_completion/dsl/rule_factory.rb

```diff
@@ -2,14 +2,15 @@ module Gitsh
   module TabCompletion
     module DSL
       class RuleFactory
-        attr_reader :root
+        attr_reader :root, :options
 
-        def initialize(root)
+        def initialize(root, options)
           @root = root
+          @options = options
         end
 
         def build(start_state)
-          root.build(start_state)
+          root.build(start_state, context: self)
```

Comment:
I suppose we could pass `options: options` instead of `context: self` here—the only thing we use the `:context` value for is to call its `#options` method.